mne/connectivity/cfc.py

```diff
@@ -0,0 +1,393 @@
+        Raw signal time series for phase calculation in PAC. If None,
+        will be the same as data_amp. Must be the same shape as data_amp.
+    n_bins : int
+        Number of phase bins. One full cycle is divided into equally-spaced
+        bins. (20 degrees * 18 = 360)
+    method : str
+        Filter method used. Default is 'fft'.
+    surrogates : bool
+        Return surrogates values for data specific to PAC,
+        the phase series and amplitude envelope are shuffled for every trial.
+        Default value is False.
+    n_jobs : int
+        Number of jobs used for parallelization. Default 1.
+
+    Returns
+    -------
+    normalized_amplitudes : ndarray, shape (n_epochs, n_bins)
+        The normalized amplitude values for the given frequency amplitude
+        across each epoch/bin.
+    phase_bins : array, shape (n_bins + 1,)
+        Binned phase points. Used for plotting.
+
+    References
+    ----------
+    Measuring phase-amplitude coupling between neuronal oscillations
+    of different frequencies.
+    Tort ABL, Komorowski R, Eichenbaum H, Kopell N., (J Neurophysiol. 2010)
+    """
+    from scipy.signal import hilbert
+
+    if data_phase is None:
+        data_phase = data_amp
+    if data_amp.shape != data_phase.shape:
+        raise AssertionError('Signals must be the same length and n_epochs.')
+
+    # obtained number of trials
+    n_epochs, n_times = data_amp.shape
+
+    # reshape epochs into single time series for filtering
+    filter_data_amp = data_amp.reshape((1, n_times * n_epochs))[0]
+    filter_data_phase = data_phase.reshape((1, n_times * n_epochs))[0]
+
+    # Filter data into phase modulating freq and amplitude modulated freq.
+    filter_x_fp = band_pass_filter(filter_data_amp, sfreq, fmin_phase,
+                                   fmax_phase, method=method, n_jobs=n_jobs)
+    filter_x_fa = band_pass_filter(filter_data_phase, sfreq, fmin_amp,
+                                   fmax_amp, method=method, n_jobs=n_jobs)
+
+    # epoch data back again after filtering
+    time_slices = np.arange(0, data_amp.size + n_times, n_times)
+    x_fa = np.zeros(data_amp.shape)
+    x_fp = np.zeros(data_amp.shape)
+    for ind in range(1, n_epochs + 1):
+        x_fa[ind - 1] = filter_x_fa[time_slices[ind - 1]: time_slices[ind]]
+        x_fp[ind - 1] = filter_x_fp[time_slices[ind - 1]: time_slices[ind]]
+
+    if len(x_fp) != len(x_fa) or x_fp.ndim != x_fa.ndim:
+        raise RuntimeError('Phase and amplitude filtered data does'
+                           ' not have same dimensions')
+
+    # Calculate phase series of phase modulating signal
+    phase_series_fp = np.angle(hilbert(x_fp)) + np.pi
+
+    # Calculate amplitude envelope of amplitude modulated signal
+    amp_envelope_fa = np.abs(hilbert(x_fa))
+
+    if surrogates is True:
+        phase_series_fp = shuffle_data(phase_series_fp)
+        amp_envelope_fa = shuffle_data(amp_envelope_fa)
```

Comment:
this is where you need to have an optional `random_state` argument to this function, that would then get passed to these shufflers